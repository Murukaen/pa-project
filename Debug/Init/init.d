Init/init.d Init/init.o: ../Init/init.c ../Init/../bitmap/bitmap.h \
 ../Init/../bitmap/../Util/util.h ../Init/../update_state/update_state.h \
 ../Init/../update_state/../move/move.h \
 ../Init/../update_state/../move/../Util/util.h \
 ../Init/../update_state/../Flags/flags.h \
 ../Init/../update_state/../Flags/../Util/util.h \
 ../Init/../update_state/../state/state.h \
 ../Init/../update_state/../state/../bitmap/bitmap.h \
 ../Init/../update_state/../state/../list/list.h \
 ../Init/../update_state/../state/../Util/util.h \
 ../Init/../state_generator/state_generator.h \
 ../Init/../state_generator/../state/state.h ../Init/../Log/log.h \
 ../Init/../Log/../state/state.h ../Init/../Log/../move/move.h \
 ../Init/../Log/../bitmap/bitmap.h ../Init/../xboard_com/xboard_com.h \
 ../Init/../Flags/flags.h ../Init/init.h

../Init/../bitmap/bitmap.h:

../Init/../bitmap/../Util/util.h:

../Init/../update_state/update_state.h:

../Init/../update_state/../move/move.h:

../Init/../update_state/../move/../Util/util.h:

../Init/../update_state/../Flags/flags.h:

../Init/../update_state/../Flags/../Util/util.h:

../Init/../update_state/../state/state.h:

../Init/../update_state/../state/../bitmap/bitmap.h:

../Init/../update_state/../state/../list/list.h:

../Init/../update_state/../state/../Util/util.h:

../Init/../state_generator/state_generator.h:

../Init/../state_generator/../state/state.h:

../Init/../Log/log.h:

../Init/../Log/../state/state.h:

../Init/../Log/../move/move.h:

../Init/../Log/../bitmap/bitmap.h:

../Init/../xboard_com/xboard_com.h:

../Init/../Flags/flags.h:

../Init/init.h:
