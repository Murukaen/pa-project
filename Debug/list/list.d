list/list.d list/list.o: ../list/list.c ../list/list.h

../list/list.h:
