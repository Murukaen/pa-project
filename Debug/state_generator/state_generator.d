state_generator/state_generator.d state_generator/state_generator.o: \
 ../state_generator/state_generator.c \
 ../state_generator/../bitmap/bitmap.h \
 ../state_generator/../bitmap/../Util/util.h \
 ../state_generator/state_generator.h ../state_generator/../state/state.h \
 ../state_generator/../state/../bitmap/bitmap.h \
 ../state_generator/../state/../list/list.h \
 ../state_generator/../state/../Util/util.h \
 ../state_generator/../read_from_file/read_from_file.h \
 ../state_generator/../read_from_file/../bitmap/bitmap.h \
 ../state_generator/../read_from_file/../state/state.h \
 ../state_generator/../read_from_file/../state_generator/state_generator.h \
 ../state_generator/../Flags/flags.h \
 ../state_generator/../Flags/../Util/util.h \
 ../state_generator/../valid_moves/valid_moves.h \
 ../state_generator/../valid_moves/../bitmap/bitmap.h \
 ../state_generator/../valid_moves/../state/state.h \
 ../state_generator/../valid_moves/../Util/util.h \
 ../state_generator/../valid_moves/../state_generator/state_generator.h \
 ../state_generator/../list/list.h ../state_generator/../Util/util.h \
 ../state_generator/../log/log.h \
 ../state_generator/../log/../state/state.h \
 ../state_generator/../log/../move/move.h \
 ../state_generator/../log/../move/../Util/util.h \
 ../state_generator/../log/../bitmap/bitmap.h

../state_generator/../bitmap/bitmap.h:

../state_generator/../bitmap/../Util/util.h:

../state_generator/state_generator.h:

../state_generator/../state/state.h:

../state_generator/../state/../bitmap/bitmap.h:

../state_generator/../state/../list/list.h:

../state_generator/../state/../Util/util.h:

../state_generator/../read_from_file/read_from_file.h:

../state_generator/../read_from_file/../bitmap/bitmap.h:

../state_generator/../read_from_file/../state/state.h:

../state_generator/../read_from_file/../state_generator/state_generator.h:

../state_generator/../Flags/flags.h:

../state_generator/../Flags/../Util/util.h:

../state_generator/../valid_moves/valid_moves.h:

../state_generator/../valid_moves/../bitmap/bitmap.h:

../state_generator/../valid_moves/../state/state.h:

../state_generator/../valid_moves/../Util/util.h:

../state_generator/../valid_moves/../state_generator/state_generator.h:

../state_generator/../list/list.h:

../state_generator/../Util/util.h:

../state_generator/../log/log.h:

../state_generator/../log/../state/state.h:

../state_generator/../log/../move/move.h:

../state_generator/../log/../move/../Util/util.h:

../state_generator/../log/../bitmap/bitmap.h:
