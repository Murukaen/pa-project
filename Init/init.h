/* General description :
 * Manages modules' initializations
 */

#ifndef _INIT_H
#define _INIT_H

/* ----- System #includes ----- */

/* ----- Local #inlcudes ----- */

/* ---- Macro #define ---- */
#define INIT_PREX 0
#define INIT_POSX 1
#define INIT_ALL 2

/* --- Types --- */

/* --- Procedures --- */

void Init ( int );

#endif

