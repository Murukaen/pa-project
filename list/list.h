/* Implementarea TAD lista simplu inlantuita
 */

#ifndef _LIST_H
#define _LIST_H

/* Types : */

#ifndef _KEY
#define _KEY
typedef void * key;
#endif

#ifndef _FFREE
#define _FFREE
typedef void ( *FFREE ) ( key );
#endif

/* FEQU - functie generica de testare a egalitatii
 *1 = true; 0 = false
 */

#ifndef _FEQU
#define _FEQU
typedef int (*FEQU) (key, key);
#endif

struct nod_lista;
typedef struct nod_lista * List;
typedef List Nod;




/* Functions : */

/*Constructor lista ( creeaza un pointer la un nod santinela - un nod de tipul nod_lista,
 * dar cu date vide )
 */
List new_list();

/* Test lista vida */
int is_empty_list(List);

/* Cauta un element intr-o lista ( prima aparitie ) .
 * Returneaza null daca nu gaseste si adresa elementului gasit in caz contrar
 */
key find_nod_list(List, key, FEQU);

/* Adauga un element la inceputul listei
 */
void add_nod_list(List, key);

/*Extragere a primului element din lista cu returnarea adresei listei ce pleaca de la urmatorul element
 * Lista va fi modificata ( primul element sters ) */
key first_nod_list(List*);

/* Destructor Lista */
void free_list(List , FFREE );

#endif

