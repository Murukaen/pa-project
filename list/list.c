#include <stdio.h>
#include <stdlib.h>
#include "list.h"


struct nod_lista {

	key data_list;
	struct nod_lista *next;

};

List new_list() {

	Nod santinela = (Nod) malloc(1 * sizeof(struct nod_lista));
	santinela -> next = NULL;
	santinela -> data_list = NULL;

	return santinela;
}

int is_empty_list(List L) {

	return L -> next == NULL;
}

key find_nod_list(List L, key content, FEQU compare_key) {

	if (L -> next == NULL) {
		return NULL;
	}

	L = L -> next;
	while (L -> next != NULL) {
				if (compare_key(L -> data_list, content) == 1) {
			return L->data_list;
		}
		L = L -> next;

	}



	if (compare_key(L -> data_list, content) == 1) {
		return L -> data_list;
	}
	return NULL;
}

void add_nod_list(List L, key content) {

	Nod new = (Nod) calloc(1, sizeof(struct nod_lista));
	new -> data_list = content;

	new -> next = L -> next;
	L -> next = new;
}

key first_nod_list(List* L) {

	if ( (*L) -> next == NULL) { // daca lista vida
		return 0;
	}

	(*L) = (*L) -> next;

	return (*L) -> data_list;
}

void free_list(List L , FFREE fre ) {

	if ( L == 0 ) return;
	List aux;
	if(L->next == NULL){
		fre ( L-> data_list );
		free(L); return;
	}
	L = L -> next;

	while(L != NULL){

		aux = L;
		L = L -> next;
		fre(aux -> data_list );
		free(aux);

	}


}

void print_list(List L) {

	if (L -> next == NULL) {
		printf("Lista vida");
		return;
	}

	while (L -> next != NULL) {
		printf("%d,", *(int*) L -> next -> data_list);
		L = L ->next;
	}
	printf("\n");

}
