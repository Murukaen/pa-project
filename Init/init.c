/* General description :
 * 
 */

/* ----- System #includes ----- */

/* ----- Local #inlcudes ----- */
#include "../bitmap/bitmap.h"
#include "../update_state/update_state.h"
#include "../state_generator/state_generator.h"
#include "../log/log.h"
#include "../xboard_com/xboard_com.h"
#include "../Flags/flags.h"
#include "init.h"

/* ---- Macro #define ---- */

/* --- Types --- */

/* --- Globals --- */

/* --- Auxiliary Procedures --- */

/* --- Procedures --- */

void Init ( int loc ) {
	
	switch ( loc ) {
		
		case INIT_PREX : 	flags_init ();
							log_init ();
							xboard_com_init ();
							break;
					
		case INIT_POSX: 	BM_init ();
							ST_gen_init ();
							break;
							
		case INIT_ALL:     	log_init ();
							flags_init ();
							Init ( INIT_POSX );
							break;
							
		case INIT_NEW:      update_state_init ();
							flags_init ();
							break;
							
	}
}
