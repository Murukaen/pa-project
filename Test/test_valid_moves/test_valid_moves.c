#include <stdio.h>
#include "../../state_generator/state_generator.h"
#include "../../valid_moves/valid_moves.h"
#include "../../list/list.h"
#include "../../Init/init.h"

int main() {

	FILE * fout = fopen("./Test/test_valid_moves/test_valid_moves.out", "w");
	FILE * fin = fopen("./Test/test_valid_moves/test_valid_moves.in", "r");

	int rez;
	STATE st;
	List l;

	Init(INIT_ALL);

	st = state_read(fin);

	P_LOC loc = LOC_new();
	LOCp_set_both(loc,4,6);


	//state_print(st, fout);

	rez = VM_is_Check_if_piece_moves(st, loc);
	printf ( "Rez:%d\n" , rez );

	return 0;
}
