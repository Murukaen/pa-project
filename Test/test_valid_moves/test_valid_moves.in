>>>Black
r n b - k p - -
p p p - - - p p
- - - r P - - -
- - - - - - - -
- - b - - - - -
- - - - - - - K
P P P - - - - P
R N B - - - N R

