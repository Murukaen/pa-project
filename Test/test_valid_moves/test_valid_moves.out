
{
--------------------------------------------------------------

~~~V_BMAP~~~ : 

 V[White]=

0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
1 1 1 1 1 1 1 1 
1 1 1 1 1 1 1 1 

 V[Black]=

1 1 1 1 1 1 1 1 
1 1 1 1 1 1 1 1 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 

 V[King]=

0 0 0 0 1 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 1 0 0 0 

 V[Queen]=

0 0 0 1 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 1 0 0 0 0 

 V[Rock]=

1 0 0 0 0 0 0 1 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
1 0 0 0 0 0 0 1 

 V[Bishop]=

0 0 1 0 0 1 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 1 0 0 1 0 0 

 V[Knight]=

0 1 0 0 0 0 1 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 1 0 0 0 0 1 0 

 V[Pawn]=

0 0 0 0 0 0 0 0 
1 1 1 1 1 1 1 1 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
1 1 1 1 1 1 1 1 
0 0 0 0 0 0 0 0 

~~~Table_What~~~:

r n b q k b n r 
p p p p p p p p 
- - - - - - - - 
- - - - - - - - 
- - - - - - - - 
- - - - - - - - 
P P P P P P P P 
R N B Q K B N R 

~~~Table_Location~~~:

Table_Location[White][King]: ( e , 1 ) 
Table_Location[White][Queen]: ( d , 1 ) 
Table_Location[White][Rock]: ( h , 1 )  ( a , 1 ) 
Table_Location[White][Bishop]: ( f , 1 )  ( c , 1 ) 
Table_Location[White][Knight]: ( g , 1 )  ( b , 1 ) 
Table_Location[White][Pawn]: ( h , 2 )  ( g , 2 )  ( f , 2 )  ( e , 2 )  ( d , 2 )  ( c , 2 )  ( b , 2 )  ( a , 2 ) 
Table_Location[Black][King]: ( e , 8 ) 
Table_Location[Black][Queen]: ( d , 8 ) 
Table_Location[Black][Rock]: ( h , 8 )  ( a , 8 ) 
Table_Location[Black][Bishop]: ( f , 8 )  ( c , 8 ) 
Table_Location[Black][Knight]: ( g , 8 )  ( b , 8 ) 
Table_Location[Black][Pawn]: ( h , 7 )  ( g , 7 )  ( f , 7 )  ( e , 7 )  ( d , 7 )  ( c , 7 )  ( b , 7 )  ( a , 7 ) 

Color on move : White

Piece to move: Pawn

Move index: 0

Current piece location : ( h , 2 )

Gen_Init : 0

En Passsant Location : Void 
--------------------------------------------------------------
}

0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 1 0 0 0 
0 0 0 0 1 0 0 0 
0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 
